storage/database/src/main/java/de/danoeh/antennapod/storage/database/ReleaseScheduleGuesser.java

```diff
@@ -122,13 +126,81 @@ private static Stats getStats(List<Date> releaseDates) {
                 daysOfWeek, daysOfMonth, mostOftenDayOfWeek, mostOftenDayOfMonth);
     }
 
+    private static List<Integer> getLargeDays(ReleaseScheduleGuesser.Stats stats, int maxDaysOff) {
+        List<Integer> largeDays = new ArrayList<>();
+        for (int i = Calendar.SUNDAY; i <= Calendar.SATURDAY; i++) {
+            if (stats.daysOfWeek[i] > maxDaysOff) {
+                largeDays.add(i);
+            }
+        }
+        return largeDays;
+    }
+
+    private static Date getUniqueDate(Date date) {
+        GregorianCalendar current = new GregorianCalendar();
+        current.setTime(date);
+        current.set(Calendar.HOUR_OF_DAY, 0);
+        current.set(Calendar.MINUTE, 0);
+        current.set(Calendar.SECOND, 0);
+        current.set(Calendar.MILLISECOND, 0);
+        return current.getTime();
+    }
+
+    private static List<Date> getMostRecentDates(List<Date> releaseDates, int amount) {
+        Set<String> uniqueDates = new HashSet<>();
+        int i = releaseDates.size();
+        while (i > 0 && amount > 0) {
+            i--;
+            String dateString = getUniqueDate(releaseDates.get(i)).toString();
+            if (!uniqueDates.contains(dateString)) {
+                uniqueDates.add(dateString);
+                amount--;
+            }
+        }
+        return releaseDates.subList(i, releaseDates.size());
+    }
+
+    private static boolean containsMultipleReleasesPerDay(List<Date> releaseDates, int threshold) {
+        Set<String> uniqueDates = new HashSet<>();
+        for (int i = 0; i < releaseDates.size(); i++) {
+            uniqueDates.add(getUniqueDate(releaseDates.get(i)).toString());
+            // The threshold indicates the number of releases need to be released on the same day for a feed to
+            // be considered a multiple-per-day podcast
+            if (i > (threshold + uniqueDates.size())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static int getAverageReleasesPerDay(List<Date> releaseDates) {
```

Comment:
Isn't `getAverageReleasesPerDay() > 1` somehow the same as `containsMultipleReleasesPerDay()`? Shouldn't an average be a floating point number instead of an integer? Alternatively, wouldn't a median be more stable than an average? That ensures that one-off days don't change the number (and works with an integer).